internal/cli/uploadcmd/upload.go

```diff
@@ -70,10 +74,24 @@ func (c *Config) Exec(ctx context.Context, args []string) error {
 		return flag.ErrHelp
 	}
 
-	if _, err := os.Stat(args[0]); os.IsNotExist(err) {
+	if _, err := os.Stat(args[0]); errors.Is(err, os.ErrNotExist) {
 		return err
 	}
 
+	// Validate the semver version constraints
+	if c.VersionConstraintsSemver != "" {
+		if _, err := version.NewConstraint(c.VersionConstraintsSemver); err != nil {
```

Comment:
Can we store this constraint somewhere? Just so that we can use it directly inside `meetsSemverConstraints()` to avoid having to call NewConstraint inside that function? 🙂

Reply:
Where would you want to store it, though? In the `Config` struct?
This is related to the comment that I raised here: https://github.com/TierMobility/boring-registry/pull/7#discussion_r644612132. So should we store both constraints in the `Config` alongside the flags? To me, it feels like we shouldn't pass on the flag variables once we've processed and parsed them. But it's also not that important, we just have to remember that the constraint is to be used, not the flag string variable.

Reply:
Ah sorry, I meant the returned [constraints](https://pkg.go.dev/github.com/hashicorp/go-version#NewConstraint), not the string flag value :)

I see your point re: storing the strings that are only used once for parsing, 
let's declare them outside of the config, agree that it's a bit weird 👍

Reply:
Sounds good, let's do that in a separate refactoring outside of the PR :)

Reply:
Nevermind, just pushed another commit!:)